lib/ansible/module_utils/basic.py

```diff
@@ -398,12 +401,12 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._verbosity = 0
         # May be used to set modifications to the environment for any
         # run_command invocation
-        self.run_command_environ_update = {}
-        self._clean = {}
+        self.run_command_environ_update: dict[str, str] = {}
+        self._clean: dict | str | None = {}
 
         self.aliases = {}
-        self._legal_inputs = []
-        self._options_context = list()
+        self._legal_inputs: list = []  # no longer used?
+        self._options_context: list = list()  # no longer used?
```

Comment:
Aren't these inferred? This is only useful if you add an item type.

Reply:
`mypy` complains about them once it checks the constructor :shrug: I guess it would prefer to have `list[xxx]` for some `xxx`, but can't figure out what `xxx` is.

Reply:
Oh, right, it can't infer element types from empty containers.
Since it seems to be unused, why not make it always-empty in typing?
```suggestion
        self._legal_inputs: list[t.Never] = []  # no longer used?
        self._options_context: list[t.Never] = []  # no longer used?
```

Reply:
`t.Never` cannot be used in module utils.